frontend/src/scenes/funnels/funnelLogic.ts

```diff
@@ -440,7 +440,7 @@ export const funnelLogic = kea<funnelLogicType>({
         interval: [() => [selectors.apiParams], (apiParams) => apiParams.interval || ''],
         stepsFromResult: [
             () => [selectors.results, selectors.timeConversionResults],
-            (trendResults, timeConversionResults) => timeConversionResults?.steps ?? trendResults ?? [],
+            (results, timeConversionResults) => (Array.isArray(results) ? results : timeConversionResults?.steps) ?? [],
```

Comment:
I'm unsure of what is the actual shape of data here – there are types, but they don't give me a ton of confidence here, e.g. why are do we use both `timeConversionResults` and `results` here, if they are both derived from `rawResults.results`. This selector seems to make pretty important assumptions about the shape of data, so I would super appreciate a comment describing these assumptions.

Reply:
They way I read this:
```ts
timeConversionResults: [
    (s) => [s.rawResults],
    ({ results, filters }): FunnelsTimeConversionBins | null =>
        filters.funnel_viz_type === FunnelVizType.TimeToConvert 
            ? (results as FunnelsTimeConversionBins) 
            : null,
],
```

... `timeConversionResults` is null if we're not on the time conversion view. So if we're not using the `steps` from `timeConversionResults ` while in the view, what are we doing with them? And the `results` should not be an array if timeConversionResults is set. 

Plus this all works for me on the app, so I'm not sure what will change here.

Reply:
That's fair, I think we should be using the steps from timeConversionResult rather than using `results` if we're in the time conversion view. Perhaps the right check here isn't an `Array.isArray()` but:

```
stepsFromResult: [
            () => [selectors.results, selectors.timeConversionResults],
            (results, timeConversionResults) => (filters.funnel_viz_type === FunnelVizType.TimeToConvert 
                        ? timeConversionResults?.steps
                        : results) ?? [],
]
```

Unfortunately, I can't seem to reproduce the specific bug I was running into, but I iirc the bug came about in the steps view where `stepResults` was always being set to `[]` because `timeConversionResults.steps ?? trendResults = []` as `[]` isn't falsy to fall back on `trendResults`. This should only happen when timeConversionResults is empty but trend results are populated.

@Twixes adding a comment is a great idea. I'll add some now.